plugins/woocommerce/src/Internal/ProductFilters/FilterData.php

```diff
@@ -0,0 +1,331 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Internal\ProductFilters;
+
+use WC_Cache_Helper;
+use Automattic\WooCommerce\Internal\ProductFilters\Interfaces\QueryClausesGenerator;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class for filter counts.
+ */
+class FilterData {
+	/**
+	 * Instance of QueryClauses.
+	 *
+	 * @var QueryClausesGenerator
+	 */
+	private $query_clauses;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param QueryClausesGenerator $query_clauses Instance of QueryClausesGenerator.
+	 */
+	public function __construct( QueryClausesGenerator $query_clauses ) {
+		$this->query_clauses = $query_clauses;
+	}
+
+	/**
+	 * Get price data for current products.
+	 *
+	 * @param array $query_vars The WP_Query arguments.
+	 * @return object
+	 */
+	public function get_filtered_price( array $query_vars ) {
+		$pre_filter_counts = $this->pre_get_filter_counts( 'price', $query_vars, );
+
+		if ( isset( $pre_filter_counts ) ) {
+			return $pre_filter_counts;
+		}
+
+		$transient_key = $this->get_transient_key( $query_vars, 'price' );
+		$cached_data   = $this->get_cache( $transient_key );
+
+		if ( ! empty( $cached_data ) ) {
+			return $cached_data;
+		}
+
+		global $wpdb;
+
+		add_filter( 'posts_clauses', array( $this->query_clauses, 'add_query_clauses' ), 10, 2 );
+		add_filter( 'posts_pre_query', '__return_empty_array' );
+
+		$query_vars['no_found_rows']  = true;
+		$query_vars['posts_per_page'] = -1;
+		$query_vars['fields']         = 'ids';
+		$query                        = new \WP_Query();
+
+		$query->query( $query_vars );
+		$product_query_sql = $query->request;
+
+		remove_filter( 'posts_clauses', array( $this->query_clauses, 'add_query_clauses' ), 10 );
+		remove_filter( 'posts_pre_query', '__return_empty_array' );
+
+		$price_filter_sql = "
+		SELECT min( min_price ) as min_price, MAX( max_price ) as max_price
+		FROM {$wpdb->wc_product_meta_lookup}
+		WHERE product_id IN ( {$product_query_sql} )
+		";
+
+		$results = $wpdb->get_row( $price_filter_sql ); // phpcs:ignore
```

Comment:
It'd be good to include a comment why this is ignored, also, does it have to be a global ignore for the line? Are you sure all the query clauses injected in via `posts_clauses` are prepared?

Reply:
I added comments explaining why we don’t use `$wpdb->prepare` for queries in this class in 701d9a706860. You’re right that it doesn’t have to be global; I changed it to `WordPress.DB.PreparedSQL.NotPrepared`.

> Are you sure all the query clauses injected in via posts_clauses are prepared?

No, it’s the plugin’s responsibility to prepare them, so we can’t say that all query clauses are prepared, so the query built by `WP_Query` in this case is not fully prepared (I was wrong about this). We can only ensure query clauses that come from WooCommerce are prepared.